numpy/_core/src/umath/loops_hyperbolic.dispatch.cpp.src

```diff
@@ -73,7 +77,145 @@
  *         achieve wider than target precision.
  *
  */
-#if NPY_SIMD_F64
+
+namespace hn = hwy::HWY_NAMESPACE;
+
+const hn::ScalableTag<float> f32;
+const hn::ScalableTag<double> f64;
+const hn::ScalableTag<int32_t> s32;
+const hn::ScalableTag<int64_t> s64;
+const hn::ScalableTag<uint32_t> u32;
+const hn::ScalableTag<uint64_t> u64;
+using vec_f32 = hn::Vec<decltype(f32)>;
+using vec_f64 = hn::Vec<decltype(f64)>;
+using vec_s32 = hn::Vec<decltype(s32)>;
+using vec_s64 = hn::Vec<decltype(s64)>;
+using vec_u32 = hn::Vec<decltype(u32)>;
+using vec_u64 = hn::Vec<decltype(u64)>;
+using opmask_t = hn::Mask<decltype(f32)>;
+
+struct hwy_f32x2 {
+    vec_f32 val[2];
```

Comment:
This is unfortunately not portable to SVE and RISC-V. We can either use Create2 to return a native tuple (caveat: this requires relatively recent compilers, check `HWY_HAVE_TUPLE`), or we can just use individual in/out function args to pass the two members.